### PR TITLE
Api Refactor

### PR DESCRIPTION
**Pull request by mistergone**
OMG, many change. Such commit. So diff.
## Additions
- AJAX requests for school data and program data
- A method of publishing many financial model changes at once by passing an object to a dispatcher (in `publish-update.js`)
## Removals
- `window` object requests for school and program data 
## Changes
- Changed app to adapt to new source of school and program data
- Changed some tests for correct debt values (`institutionalLoanRate` was not being imported from the URL properly, and I had accidentally typed the same incorrect rate into the spreadsheet)
## Testing
- You shouldn't need to `npm install` or anything, but you might as well!
## Review
- @marteki @niqjohnson @higs4281 
## Todos
- Oops, I left in some commented-out old code. Ought to remove that.
## Checklist
- [x] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
- [x] Passes all existing automated tests
- [x] New functions include new tests
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged
- [ ] Visually tested in supported browsers and devices 
- [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
## Special Thanks
- @niqjohnson, for closing a PR that would've made more conflicts. :smile: 
## Animated GIF

When I refactor code, I'm like...
groucho-dance

### CODE REVIEW

File: src/disclosures/js/index.js [hunk 1]
```diff
@@ -3,10 +3,13 @@
 var fetch = require( './dispatchers/get-api-values' );
 var financialModel = require( './models/financial-model' );
 var schoolModel = require( './models/school-model' );
+var getModelValues = require( './dispatchers/get-model-values' );
+var getUrlValues = require( './dispatchers/get-url-values' );
```
Comment (marteki) [inline src/disclosures/js/index.js:7]: I think the PR is missing the `get-url-values` file.

Comment (mistergone) [inline src/disclosures/js/index.js:7]: Yep, I realized I forgot it in the shower this morning, and I was hoping to sneak it in. Coming soon!

**Comment by niqjohnson** [comment]
I think this might be a known issue, but things broke down for me when testing for a school without program data (i.e. any non-settlement school that either doesn't have a `pid` in the URL or has a `pid` that doesn't return any program data).

Is figuring that out out of scope for this PR? The big effect this issue was having in my testing is I can't verify if graphs are working right for schools without program data.

**Comment by mistergone** [comment]
@niqjohnson - I would call that "out of scope" if only to try to keep this PR from languishing.

Can you maybe file an issue that describes what the expected behavior would be, and I'll work on it?

**Comment by niqjohnson** [comment]
Cool, issue filed: #120 

Everything seems to be working for me for our test school with program data, so :+1: there. Nice!

If this PR gets merged before #117, I'll add a note there that testing the graphs without program data won't work until #120 is closed.

**Comment by mistergone** [comment]
Merging this now...

**Pull request merged**